hera_cal/reflections.py

```diff
@@ -267,25 +228,439 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
             if cal.Ntimes > self.Ntimes:
```

Comment:
Something is weird here. Why would they not match? It seems to me like if the number of times and/or the number of freqs doesn't match, it's an error.

Reply:
There are cases where a calibration is a single integration that we want to broadcast across all times, or a single channel that we want to broadcast across all freqs. In the case where the broadcasting isn't allowed, it will error in `io.write_cal`.

Reply:
Is that something we ever want to do? Or do we want to force the user to do the broadcasting rather than do it silently?

Reply:
The broadcasting is something I do routinely: my CASA calibration solutions are single spectra gains, so anytime I apply the CASA gains using `apply_cal` I rely on this broadcasting ability.